### ISSUE TITLE
[17.0] [18.0] STORNO invoicing

### ISSUE THREAD

**Issue by gezzzan** [hungarodo/odoohu-plus#32]
currently it is not possible to issue STORNO invoice

it is essential in hungarian invoicing practice

17.0: https://github.com/odoo/odoo/blob/8749d01b9f4fe7dcc761b2ad1cb7c160b674dc0a/addons/l10n_hu_edi/models/account_move.py#L516

18.0: https://github.com/odoo/odoo/blob/ec5f63117e1a217b0a91ac4358aea6b50e423135/addons/l10n_hu_edi/models/account_move.py#L513

NAV Online Invoice specification: 2.5.1 Data report on invoice cancellation https://onlineszamla.nav.gov.hu/files/container/download/Online%20Invoice%20System%203.0%20Interface%20Specification.pdf

**Comment by odootechnology** [comment]
The Hungarian law about VAT ([2007. évi CXXVII. törvény az általános forgalmi adóról](https://net.jogtar.hu/jogszabaly?docid=a0700127.tv)) makes diffference between the STORNO and MODIFICATION invoice.
Storno/Cancellation invoice in Hungarian "Érvénytelenítő számla" and the Modification invoice is "Módosító számla"
see the following paragraphs: 
- § 55. (2) b.)
- § 77  (1)
- § 78 (1) a.)
- § 153/B  (1) a.)
- § 153/C  (1) a.)
- § 196/B  (2) c.) and ca.) and cc.)
 

Regarding to the TAX law the [Online Invoice System 3.0 Interface Specification] (https://onlineszamla.nav.gov.hu/api/files/container/download/Online%20Invoice%20System%203.0%20Interface%20Specification.pdf) (shortly: OIS 3.0) also expect the differences between the Storno/Cancellation (STORNO) and Modification (MODIFY) invoice, 
see the OIS 3.0 specification 
"2.5.1 Data report on invoice cancellation" and 
"2.5.2 Data report on invoice modification" sections !!!!

Storno invoice cancel the whole invoice, there is no modification.
Modify invoice just modifying some part of the invoice.
So in Odoo when the user want to create a credit note, the user has to decide which one has to be done.
In Odoo version 16 and before there was an option at "add credit note" function: "full refund" (Cancel invoice) or "Partial refund" (modify) option.
The full refund function was a cancellation/ storno function. We would need something like this in the Hungarian localisation.

Summing up, there is a difference between a Storno/cancell invoice and a Modify invoice, and has to add the right value in the XML report.

**Comment by Eilerant** [comment]
Thank you !

Your ticket ID is #4577553. You'll soon receive an email confirming the reception of your ticket.
https://www.odoo.com/my/tasks/4577553?search_in=name&search=4577553

To provide additional information, you can simply reply to the confirmation email.

Please also note that you can access all your support tickets at any time in [your user portal](https://www.odoo.com/my).

**Comment by gezzzan** [comment]
18.0 commit: https://github.com/odoo/odoo/commit/5214296e8be76663ffdb6647c91645c66501121d

**Comment by gezzzan** [comment]
made some test, looks ok
